Jar design ceg4110

JoshRupp opened the pull request.

Kazalor committed:
Update Actions.md

Updated the Action list, added 12, 13, 14, 15

JoshRupp committed:
Merge branch 'develop' of https://github.com/EnderStir/ceg4110-group-12-spring-2021 into main

JoshRupp committed:
Finished my Design.md file. Need it to be PR'ed so we can proceed with
completion of this aspect of the project.  Moreover, started creating
a commands moduele  this morning, April 15, 2021. Hoping we can utilize
this code as an abstract dictionary list of commands and its functions.

EnderStir commented on Information/Design.md:
```diff
@@ -89,3 +89,320 @@
+
+[Five requirements of Group 12's Google Doc which can be found at:]
+ 
+(https://docs.google.com/document/d/1LdCnvcZ_LXHHckbBnyPZtQZOH8F8qjPd47igwdDOoVs/edit)
```
Google has permissions on files, so we may need to add Buck's account to this file if we want him to see it. Also, all of the content of this file should be moved to the Requirements.md file in the Repo. So we should not need this link at all.

JoshRupp replied:
Ok, I will remove it.

EnderStir commented on Information/Design.md:
```diff
@@ -89,3 +89,320 @@
+ 
+(https://docs.google.com/document/d/1LdCnvcZ_LXHHckbBnyPZtQZOH8F8qjPd47igwdDOoVs/edit)
+
+There are 18 total requirements for this project for building and implementing a _Discord
```
The number of requriement will change, so don't state this number in the design, as it will be wrong. Actually, it is already wrong.

JoshRupp replied:
LOL, understood.

EnderStir commented on Information/Design.md:
```diff
@@ -89,3 +89,320 @@
+(https://docs.google.com/document/d/1LdCnvcZ_LXHHckbBnyPZtQZOH8F8qjPd47igwdDOoVs/edit)
+
+There are 18 total requirements for this project for building and implementing a _Discord
+bot_ for the _Monster Hunter World_ Video Game. This document specifically focuses on the 
```
The rest of this paragraph should be at the top of the document.

JoshRupp replied:
OK, will do

EnderStir commented on Information/Design.md:
```diff
@@ -89,3 +89,320 @@
+bot_ for the _Monster Hunter World_ Video Game. This document specifically focuses on the 
+description, of the design process, for the following requirements. 
+
+**REQ 130** 
+
+_The bot shall be able to provide the recipe for the craft-able item requested by the user_
+
+
+**REQ 140**
+
+_The bot shall allow privileged users add data to the database_
+
+
+**REQ 150**
+
+
+_The bot shall allow privileged users edit data to the database_
+
+
+
+**REQ 160**
+
+_The bot shall allow users to track the monsters they have killed_
+
+
+**REQ 170**
+
+_The bot shall allow users to track the monsters that have killed them_
+
+
+**REQ 180**
+
+_The bot shall be able to display a list of all available commands supported by the bot_
+ 
+
+------------------------------------------------------------------------------------------
```
Don't restate requirements in a design document, as they will get out of sync.

JoshRupp replied:
Ok, I was attempting to explicitly restate the REQ for each of my designs for the commands. In attempt to make it easy for the professor and the TA to track from REQ --> design. However, I see your point. Looks tacky potentially could get unorganized.   Will remove.

EnderStir commented on Information/Design.md:
```diff
@@ -89,3 +89,320 @@
+
+
+
+#Design.md by Joshua Rupp
+
+######CEG 4110 Spring 2021 
+
+######Joshua Rupp, Kyle Sturdevant, Trenton Brown --> Group 12
+
+######Overview of what you will find in this document:
```
Adding a Header in the middle of the file is bad form. This should be at the top of the file. And the file is not just by you, as we are all contributing to the file. So it should be "by Group 12" if we need this at all.

JoshRupp replied:
Ok. I will take it out.

EnderStir commented on Information/Design.md:
```diff
@@ -89,3 +89,320 @@
+######REQ 130 Design: 
+
+
+Command name: _-item make_
```
Per our convention commands should be one word. So just "-item"

JoshRupp replied:
Ok will adapt

EnderStir commented on Information/Design.md:
```diff
@@ -89,3 +89,320 @@
+
+Command name: _-item make_
+
+Parameters: _name_, _type_, _rarity_ 
```
This is unclear, do you need all 3 parameters or just one?

JoshRupp replied:
Um all three could be useful.  For instance, if someone searches for an item he/she would like to see if its craftable.  If the name of that item is misspelled, but they get the type correct then the command still produces the correct result.

EnderStir commented on Information/Design.md:
```diff
@@ -89,3 +89,320 @@
+**Design of _item make_ command**:
+
+The name for the command, _item_ ,was a straight forward process. I needed a name that would 
+non ambiguous.  In contrast, if the command name was _-i_ or _-itm_. There leaves a level of 
```
I believe you are missing the word "be" after "would"

JoshRupp replied:
ok will edit

EnderStir commented on Information/Design.md:
```diff
@@ -89,3 +89,320 @@
+The name for the command, _item_ ,was a straight forward process. I needed a name that would 
+non ambiguous.  In contrast, if the command name was _-i_ or _-itm_. There leaves a level of 
+decoding from the users perspective, on trying to figure out the function of the command. 
+Where as _-item make_ does not require a significant amount more typing. Yet leaves no ambiguity
```
Above you say it is "-item" and here you say it is "-item make". Inconsistent.

JoshRupp replied:
Ok will edit

EnderStir commented on Information/Design.md:
```diff
@@ -89,3 +89,320 @@
+this command does require that the user needs to know exactly one of the three ways of 
+quantifying an item in the Monster Hunter World game. Providing no parameters, returns 
+nothing, except a message telling the user to input a name, type, or rarity if they wish
+to see a craft-able recipe for that item, if applicable. Not all items in the Monster 
```
Spelling: 'craftable' is a word. craft-able is not a word.

JoshRupp replied:
Ok will look into it

EnderStir commented on Information/Design.md:
```diff
@@ -89,3 +89,320 @@
+
+examples of _item make_ command usage: 
+
+`-item make Mega Potion`  -- example with _name_ parameter 
+
+`-item make medicine`     -- example with _type_ parameter
```
How do you tell the difference between a name and type if they are both strings?

JoshRupp replied:
In the implementation process, you should not compare either or of name, or type parameter.  IF the user supplies only one or the other, then the command will still work. If the user provides both of them.  Then it can either use one or the other, or user both, in case one produces a result and one does not.  Moreover, if the user miss spells one or the other than the command can still produce an output of desired results.

EnderStir commented on Information/Design.md:
```diff
@@ -89,3 +89,320 @@
+######REQ 140 Design:
+
+
+Command name: _-user_ 
```
Shouldn't this be an 'add' command?

JoshRupp replied:
Yes, in my first attempt to design this command. It was not as good as it could have been. I have adjusted the design of this command. It will show on in the following pushes to this branch.

EnderStir commented on Information/Design.md:
```diff
@@ -89,3 +89,320 @@
+######REQ 150 Design:
+
+
+Command name: _-user_ 
```
Shouldn't this be 'edit'? How do we tell the difference between this -user and the other -user?

JoshRupp replied:
Yea these command's where not designed well initially.y I blame that on lack of design experience. Moreover, on stress do to many classes and homework volume. Which ever the case of excuse, this design will be adjusted.

EnderStir commented on Information/Design.md:
```diff
@@ -89,3 +89,320 @@
+######REQ 160 Design:
+
+
+Command name: _-my victims_ 
```
Commands should be one word.

JoshRupp replied:
Ok, I will change them to reflect that. That is a good point.

EnderStir commented on Information/Design.md:
```diff
@@ -89,3 +89,320 @@
+The command _my victims_ takes in the name of a monster in the monster hunter world video
+game, as its only parameter. It installs it into a list of monsters, and then displays 
+the list to the user via discord. If the name of the monster is not a monster in the 
+Monster Hunter World game. Then the Bot prints out an appropriate message to the user. IF
+the command is entered without any name, or parameter. Then command does nothing and prints
+out nothing to the user via Discord. 
```
If the command isn't given the proper parameters, it should print that commands 'help'.

JoshRupp replied:
Very good point. I will reflect that in my design.

EnderStir commented on Information/Design.md:
```diff
@@ -89,3 +89,320 @@
+When this command is entered, it will first prompt the user for the user's account name. This 
+way each user has its own saved list of monsters killed. If the user does not have a valid 
+account name or id then the monster is not saved in any list, and an appropriate message is 
+displayed to the user via discord. The user will have to authenticate themselves via their 
+password as well. This will be the same password as in the Discord server. This way random 
+people belonging to the server this Monster Hunter World bot is associated with. Cannot 
+edit the list of monsters they have killed without the knowledge of the users pass code.
```
We can tell the user that ran the command, why would we need them to login?

EnderStir commented on Information/Design.md:
```diff
@@ -89,3 +89,320 @@
+######REQ 170 Design:
+
+
+Command name: _-gone to soon_ 
```
Command can only be one word.

JoshRupp replied:
OK, I will change that design aspect.

EnderStir commented on Information/Design.md:
```diff
@@ -89,3 +89,320 @@
+
+Parameters: _name_ 
+
+**Design of _user_ command**:
```
This is not the 'user' command

JoshRupp replied:
LOL, I believe that I copy and pasted thinking I would edit the command name to reflect each command sections when authoring this Design.md document. However, I overlooked it, apparently.

EnderStir commented on Information/Design.md:
```diff
@@ -89,3 +89,320 @@
+
+Parameters: _name_ 
+
+**Design of _user_ command**:
```
This is not the 'user' command

JoshRupp replied:
LOL, yea I saw this when editing the file just now. I was perplexed at how this occurred as well.

EnderStir commented on Information/Design.md:
```diff
@@ -89,3 +89,320 @@
+Discord Server. The key in the dictionary is the command name, the value shall be a brief 
+description or purpose of the command.  
+
+In total, the number of commands are 18 for this discord bot. Thus the size of this data 
+structure is 18 cells in the command dictionary. 
```
The number of commands may change, don't specificy size. Also python doesn't require size defined, so why would we care?

JoshRupp replied:
Ok, I have changed this aspect of the design already. I molded your design advice, with mine. I hope it is ok.

EnderStir commented on Information/Design.md:
```diff
@@ -89,3 +89,320 @@
+people belonging to the server this Monster Hunter World bot is associated with. Cannot 
+edit the list of monsters they have killed without the knowledge of the users pass code.
+
+There are 48 total monsters in the Monster Hunter World video game. Thus the maximum size 
+of any monster list will be 48. The smallest size of course would be if there were no 
+monsters killed, thus size = 0. Being how the upper bound of a monster list is 48. 
+printing out to discord should not cause any significant undefined bandwidth issues. 
```
The number of monsters may change, don't specify this in the design.

JoshRupp replied:
Ok, I will take a look at it and try my best to adjust it in an educated way.

EnderStir commented on Information/Design.md:
```diff
@@ -89,3 +89,320 @@
+`-my victims Kulu-Ya-Ku`  -- example with _name_ parameter 
+
+
+example of the _user entry_ aspect of the _my victims_ command:
+
+`Enter user account name: Josh`
+
+`Enter the password for Josh: ********`
```
This doesn't appear to be required. As we can tell what user ran the command, so we will just add the monster to that user's list.

JoshRupp replied:
Ok, I will modify the design

EnderStir commented on Information/Design.md:
```diff
@@ -89,3 +89,320 @@
+`-gone to soon Kulu-Ya-Ku`  -- example with _name_ parameter 
+
+
+example of the _user entry_ aspect of the _gone to soon_ command:
+
+`Enter user account name: Josh`
+
+`Enter the password for Josh: ********`
```
This also doesn't appear to be required, due to same reasons as REQ 160

JoshRupp replied:
dido to ^^

EnderStir commented on commands.py:
```diff
@@ -0,0 +1,119 @@
+
+
+
+
+
+"""   
+This function assumes that the Parameter is a single, non nested, dictionary of strings called printable_cmd_list.
+
+ Key of any member of the dictionary printable_cmd_list is command name of Monster Hunter World Discord Bot.
+
+ Value of any member of the dictionary printable_cmd_list is brief description of command name for the Monster Hunter
+ World Discord Bot.
+
+ return value none. --> Prints the contents of printable dictionary to stdout. 
+
+ Prints contents of dictionary printable_cmd_list to stdout. Helps illustrate the available commands to the Monster
+ Hunter World Discord Bot.
+
+"""
+def help(self):
+
+    str1 = "Commands:"
+    str2 = "Description:"
+    print(str1.ljust(20, " "), str2.rjust(20, " "))
+
+    for key, value in self.printable_cmd_list.items():
+        print(key.ljust(20, ""))
+
+
+
+
+
+
+
+
+
+
+
+
+
```
This entire file appears to add unneeded complexity. We do not need a programmatic way to add commands to the bot. The commands will be coded directly into the bot.

JoshRupp replied:
Ok

EnderStir commented on Information/Design.md:
```diff
@@ -89,3 +89,320 @@
+------------------------------------------------------------------------------------------
+
+
+######REQ 130 Design: 
```
Please move this entire design section above the "Backend" section, so all of the REQs are together and in order.

JoshRupp replied:
Understood

EnderStir reviewed (changes requested):
Actions file is fine. Other files have corrections.

JoshRupp committed:
Implemented basic version of printing a formatted table.

JoshRupp committed:
Merge branch 'develop' of https://github.com/EnderStir/ceg4110-group-12-spring-2021 into feature/Josh-Action23-printTable

JoshRupp committed:
I was stupid and pushed the printTables.py file into wrong directory.
Had to correct. Hope no judgment is passed my way.

JoshRupp committed:
few additional changes for resizing the basic version of the
printTables.py

JoshRupp committed:
few additional changes for resizing the basic version of the
printTables.py

JoshRupp committed:
Corrections to  Design.md as stated by EnderStir from PR comments.

JoshRupp committed:
Design.md updates from the instructions to fix it, from the open PR.

JoshRupp committed:
Converted our Minutes Log file from our group 12 google doc to a
(.md) file, and deposit it to our ceg 4110 repository.

JoshRupp committed:
Update Design.md

I fixed the changes in the comments to the best of my knowledge. If upon further review there needs to be better adjustments please advise.

JoshRupp committed:
Delete commands.py

JoshRupp committed:
pulled from develop branch. Had to commit these changes

JoshRupp committed:
Merge branch 'jarDesignCEG4110' of https://github.com/EnderStir/ceg4110-group-12-spring-2021 into jarDesignCEG4110

JoshRupp committed:
resolved the merge conflict manually with the Actions.md file.
Edited my Design.md file a little more and should be consistent with the
recommended changes by the reviewer of the current PR.

JoshRupp committed:
Adjusted my Design.md file to reflect the input of the reviewer of my
PR for this file.

JoshRupp committed:
Updated minutes.md, Design.md.  To the most current versions.

JoshRupp committed:
Updated minutesLog.md

JoshRupp committed:
Merge pull request #11 from EnderStir/minutesLog/Josh-Action25-convertedGoogleDoc

Minutes log/josh action25 converted google doc

JoshRupp committed:
Merge branch 'develop' of https://github.com/EnderStir/ceg4110-group-12-spring-2021 into jarDesignCEG4110

# Conflicts:
#	Information/Design.md

JoshRupp committed:
updated Design.md and added a requirement to the Requirements.md file.

JoshRupp committed:
updated Design.md

EnderStir reviewed (commented):
Looks good to merge.
